src/devices/bus/gameboy/mbc.cpp

```diff
@@ -1439,3 +1449,117 @@ void gb_rom_camera_device::write_ram(offs_t offset, uint8_t data)
 			m_ram[ram_bank_map[m_ram_bank] * 0x2000 + (offset & 0x1fff)] = data;
 	}
 }
+
+// Vast Fame
+
+uint8_t gb_rom_vfame_device::read_rom(offs_t offset)
+{
+	bool inchargestartbank = (m_seq_start_bank == 0 && offset < 0x4000) || (m_seq_start_bank == m_latch_bank2 && offset >= 0x4000);
+	
+	if (inchargestartbank && offset == m_seq_start_addr && m_seq_bytes_left == 0)
+	{
+		m_seq_bytes_left = m_seq_length;
+	}
+	
+	if(m_seq_bytes_left > 0 && offset < 0x8000)
+	{
+		m_seq_bytes_left--;
+		return m_seq[m_seq_length - m_seq_bytes_left - 1];
+	}
+	
+	if(m_shouldreplace && offset >= m_replace_start_addr && offset < 0x4000)
+	{
+		return m_rom[m_replace_src_bank * 0x4000 + (offset & 0x3fff)];
+	}
+
+	if (offset < 0x4000)
+		return m_rom[rom_bank_map[m_latch_bank] * 0x4000 + (offset & 0x3fff)];
+	else
+		return m_rom[rom_bank_map[m_latch_bank2] * 0x4000 + (offset & 0x3fff)];
+}
+
+void gb_rom_vfame_device::write_bank(offs_t offset, uint8_t data)
+{
+	if (offset < 0x2000)
+		m_ram_enable = ((data & 0x0f) == 0x0a) ? 1 : 0;
+	else if (offset < 0x3000)
+	{
+		// MBC5 has a 9 bit bank select
+		// Writing into 2000-2fff sets the lower 8 bits
+		m_latch_bank2 = (m_latch_bank2 & 0x100) | data;
+	}
+	else if (offset < 0x4000)
+	{
+		// MBC5 has a 9 bit bank select
+		// Writing into 3000-3fff sets the 9th bit
+		m_latch_bank2 = (m_latch_bank2 & 0xff) | ((data & 0x01) << 8);
+	}
+	else if (offset < 0x6000)
+	{
+		data &= 0x0f;
+		if (has_rumble)
+		{
+			m_rumble = BIT(data, 3);
+			data &= 0x7;
+		}
+		m_ram_bank = data;
+	}
+	else if (offset < 0x8000)
```

Comment:
Likewise everything above here is the parent class function. Something like `if (offset >= 0x6000) // do new stuff else // call parent write_bank` would work.